src/actions/checkBalance.ts

```diff
@@ -141,7 +141,7 @@ async function deriveAndScanAddressesByDerivationMode(
         txCounter++;
         display.transientLine(/* delete address as it is not an active one */);
 
-        if (account === 1 || txCounter >= configuration.gap_limit) {
+        if (account === 1 || txCounter >= +configuration.gap_limit) {
```

Comment:
a bit shady/hacky to use "unary plus" operator there (and that's suprising that Typescript allows comparing Number & String here, but probably that `configuration` typing is broken)

would rather recommend to directly cast as Number in `src/configuration/settings.ts` or to explicitely cast as Number in place here

Reply:
Thank you @meriadec: I have addresses your concern. You can merge if the changes suit you (I am not authorized to merge by myself).